### ISSUE TITLE
MODTraitsXD3 seems compatible with xD4 (dragon's dogma: dark arisen) in mod_to_gltf

### ISSUE THREAD

**Issue by RamKromberg** [PredatorCZ/RevilLib#2]
When I run "revil_toolset.exe mod_to_gltf ddda_foobar.mod" it naturally complained about version 212 (xD4) being invalid. However, I figured if re6 is supported so there shouldn't be that much of a difference... So, after [looking at the code]( https://github.com/PredatorCZ/RevilLib/blob/2c328f3a9d6d1671c107c6582945d9fbe3cde0c0/include/revil/mod.hpp#L28), I just used HxD to hex edit the version number in the header from xD4 to xD3 (re6's) and it successfully outputted the .glb which blender loaded successfully.

Anyhow, having model output is pretty useful even without being able to put it back in for texturing as it's possible to use blender to output a uvmap for some gimp touchups or output an .obj and load it with armorpaint or the likes... So, would you consider enabling it for the toolset releases?

p.s.
Git for Windows comes with bash and dd so I just smoothed out the processes by tossing 
[this as "ddda_mod_to_gltf.sh"](https://github.com/PredatorCZ/RevilLib/files/12613893/ddda_mod_to_gltf.sh.txt) along with RevilToolset-v2.5.118-win64 in my Downloads folder as follows:

```
#!/bin/sh

#change this to point to where you extracted revil_toolset
revil_toolset_path="$HOME/Downloads/RevilToolset-v2.5.118-win64"
#e.g.
#revil_toolset_path="/C/Users/your_user_name/Downloads/RevilToolset-v2.5.118-win64"
export PATH=$PATH:"$revil_toolset_path"

where.exe revil_toolset.exe > /dev/null 2>&1
if [ $? -eq 1 ]; then
	echo ERROR: revil_toolset.exe was not found in path.
	echo Please modify this shell script to point to where you extracted the toolset before draging and dropping the .mod to this script.
	echo Press any key to quit...
	read nm
	exit
fi

if [[ $# -eq 0 ]] ; then
	echo ERROR: no argument.
	echo Please drag and drop a .mod to the shell script.
	echo Press any key to quit...
	read nm
	exit
fi

echo Backing up $1 as $1.bak...
cp "$1" "$1.bak"
echo "Which header would you like to use?"
select yn in "DD:DA" "RE6" "Cancel"; do
	case $yn in
		DD:DA ) printf '\x4D\x4F\x44\x00\xD4\x00' | dd of=$1 bs=1 seek=0 count=6 conv=notrunc; revil_toolset.exe mod_to_gltf $1; break;;
		RE6 ) printf '\x4D\x4F\x44\x00\xD3\x00' | dd of=$1 bs=1 seek=0 count=6 conv=notrunc; revil_toolset.exe mod_to_gltf $1; break;;
		Cancel ) exit;;
	esac
done
echo Restoring $1.bak as $1...
mv "$1.bak" "$1"

echo Press any key to quit...
read nm

```
That lets me simply drag & drop my .mod, hit 2 and enter to get the glb. I also left the ddda header use as a test under 1 for reference.

**Comment by PredatorCZ** [comment]
Not all models are supported for convert as of now since nobody requested those versions from me.
It's still WIP.

**Comment by PredatorCZ** [comment]
> I'm surprised seeing how it was left without model editing tools for over a decade.

I believe there are some private(?) tools used by modding communities. Thankfully MOD format is fairly simple.

From notes I have, xD3 is incompatible with other versions, but not by much.

**Comment by RamKromberg** [comment]
> I believe there are some private(?) tools used by modding communities. Thankfully MOD format is fairly simple.

I haven't looked much but the stuff I did see were model swaps and re-texturing so I assumed no one figured out how to do it...

> From notes I have, xD3 is incompatible with other versions, but not by much.

No idea. I only did a single re-texture job around a clipping polygon since it was in my face whenever an NPC walked by and that was that. Of course, if I had a real way to move the node I would have done that... But it's good enough for me to get preoccupied by the game mechanics and save game editing so I guess that was that :D

Anyhow, since you're aware of the state of things I'll close the issue.
